numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -653,6 +653,38 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_repr(PyObject *self){
+    PyVoidScalarObject *s = (PyVoidScalarObject*) self;
+    PyObject *bytes, *str;
+    if (PyDataType_HASFIELDS(s->descr)) {
+        return gentype_repr(self);
+    }
+    bytes = gentype_generic_method(self, NULL, NULL, "tobytes");
```

Comment:
you can just call `PyBytes_FromStringAndSize` on obval here (it is defined in our compat headers to work in python2 too)